impls/src/adapters/emoji.rs

```diff
@@ -156,24 +145,17 @@ impl Header {
 		// Get the "Decoder" reversing the "Encoder" so we have String_Emoji -> Header
 		let emoji2method = invert_hashmap(&method2emoji);
 
-		// For each emoji
-		let emojis = emoji_string.chars();
+		let version_str = emoji2method.get(&emoji_string).unwrap().to_owned();
 
-		// Save all emojis converted into a vector -> Vec -> Header
-		let mut header_vec = Vec::new();
+		// Get the version from str
+		let version = match version_str.parse() {
+			Ok(version) => version,
+			Err(_) => panic!("Invalid version number!"),
+		};
```

Comment:
In this function, you can improve the syntax by using:
```suggestion
		let version: u16 = version_str
			.parse()
			.unwrap_or_else(|_| panic!("Invalid version number!"));
```